tests/unit/logic/vote.spec.js

```diff
@@ -0,0 +1,1012 @@
+
+			beforeEach(function () {
+				callback = sinon.spy()
+				clock = sinon.useFakeTimers()
+				schema = new zSchema()
+				vote = new Vote(logger, schema)
+				account = { merge: function () {} }
+				sinon.stub(account, 'merge').callsFake(function (address, data, cb) {
+					cb()
+				})
+				Vote.__set__('self.scope', {account: account})
+				rounds = {calc: function (height) { return height }}
+				sender = {address: 123}
+				Vote.__set__('setImmediate', setImmediate)
+
+				vote.bind(delegates, rounds, system)
+			})
+
+			afterEach(function () {
+				clock.restore()
+			})
+
+			context('if trs.asset.votes is null', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: null}}
+					vote.undoUnconfirmed(trs, sender, callback)
+				})
+
+				it('should not call to scope.account.merge()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.false
+					expect(callback.called).to.be.true
+				})
+			})
+
+			context('if trs.asset.votes is empty', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: []}}
+					vote.undoUnconfirmed(trs, sender, callback)
+				})
+
+				it('should not call to scope.account.merge()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.false
+					expect(callback.called).to.be.true
+				})
+			})
+
+			context('if trs.asset.votes is undefined', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: undefined}}
+					vote.undoUnconfirmed(trs, sender, callback)
+				})
+
+				it('should not call to scope.account.merge()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.false
+					expect(callback.called).to.be.true
+				})
+			})
+
+			context('if trs.asset.votes has not a valid format', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: [1,2]}}
+					vote.undoUnconfirmed(trs, sender, callback)
+				})
+
+				it('should not call to scope.account.merge()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.false
+					expect(callback.called).to.be.true
+				})
+			})
+
+			context('if trs.asset.votes is an Array and have items', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: ['+37b6d3fc89fb418fa5e3799e760cc2b1733a567daebb08a531d3d7e8b24f2d22', '+b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9']}}
+					vote.undoUnconfirmed(trs, sender, callback)
+				})
+
+				it('should call to scope.account.merge() and callback()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.true
+					expect(callback.called).to.be.true
+				})
+			})
+		})
+
```

Comment:
`Vote.prototype.schema` is untested :(

Reply:
I thought that only we have to test functions

Reply:
I don't think that to test a static object is within the testing standards. What do you think guys? Anyway I'm testing Vote.prototype.objectNormalize() in deep and this function depends on Vote.prototype.schema directly.